x-pack/plugins/painless_lab/server/routes/api/execute.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../types';
+import { API_BASE_PATH } from '../../../common/constants';
+import { isEsError } from '../../lib';
+
+const bodySchema = schema.object({
+  script: schema.object({
+    source: schema.string(),
+  }),
+});
+
+export function registerExecuteRoute({ router, license }: RouteDependencies) {
+  router.post(
+    {
+      path: `${API_BASE_PATH}/execute`,
+      validate: {
+        body: bodySchema,
+      },
+    },
+    license.guardApiRoute(async (ctx, req, res) => {
+      const body = req.body;
+
+      try {
+        const callAsCurrentUser = ctx.core.elasticsearch.dataClient.callAsCurrentUser;
+
+        const response = await callAsCurrentUser('scriptsPainlessExecute', {
+          body,
+        });
+
+        return res.ok({
+          body: response,
+        });
+      } catch (e) {
+        // Invalid painless script was submitted
+        // Return 200 with error object
+        if (e.body) {
+          return res.ok({
+            body: e.body,
+          });
+        }
+
+        if (isEsError(e)) {
+          return res.customError({ statusCode: e.statusCode, body: e });
```

Comment:
I'm just wondering whether we can't collapse this section into this if check:

```ts
	// Invalid painless script was submitted
        // Return 200 with error object
        if (isEsError(e)) {
          return res.ok({ body: e.body });
        }
```

I'm guessing if we detect an ES error here it means that something went wrong with the call to painless script execution endpoint.

Reply:
That's a valid point! Fixed.